los_keeper_ros2/src/los_server/los_server.cc

```diff
@@ -2,81 +2,31 @@
 
 using namespace los_keeper;
 
-std::string LosServer::ProcessMessage(const std::string &raw_string) {
-  std::string result_string;
-  for (char c : raw_string) {
-    result_string.push_back(c);
-    result_string += ' ';
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-  }
-  return result_string;
-}
-
-void LosServer::ShortTopicCallback(const std_msgs::msg::String::SharedPtr msg) {
-  auto current_time = now();
-  std::string received_string =
-      msg->data + " (" + std::to_string(current_time.seconds()) + ")";
-  auto processed_string = ProcessMessage(received_string);
-  {
-    std::unique_lock<std::mutex> lock(mutex_list_.short_topic, std::defer_lock);
-    if (lock.try_lock()) {
-      RCLCPP_INFO(this->get_logger(), "Could set short message: %s",
-                  processed_string.c_str());
-      this->wrapper_.SetShortString(processed_string);
-    } else {
-      RCLCPP_INFO(this->get_logger(),
-                  "Unlock failed when setting short message: %s",
-                  processed_string.c_str());
-    }
-  }
-};
-void LosServer::LongTopicCallback(const std_msgs::msg::String::SharedPtr msg) {
-  auto current_time = now();
-  std::string received_string =
-      msg->data + " (" + std::to_string(current_time.seconds()) + ")";
-  auto processed_string = ProcessMessage(received_string);
-  {
-    std::unique_lock<std::mutex> lock(mutex_list_.long_topic, std::defer_lock);
-    if (lock.try_lock()) {
-      RCLCPP_INFO(this->get_logger(), "Could set long message: %s",
-                  processed_string.c_str());
-      this->wrapper_.SetLongString(processed_string);
-    } else {
-      RCLCPP_INFO(this->get_logger(),
-                  "Unlock failed when setting long message: %s",
-                  processed_string.c_str());
-    }
-  }
-};
-
-bool LosServer::Update() { return wrapper_.Plan(); }
 void LosServer::TimerCallback() {
-  std::scoped_lock<std::mutex, std::mutex> lock(mutex_list_.long_topic,
-                                                mutex_list_.short_topic);
-  if (Update()) {
-    auto concat_string = wrapper_.GetConcatString();
-    RCLCPP_INFO(this->get_logger(), "Concat string = %s",
-                concat_string.c_str());
-  }
+  std::scoped_lock<std::mutex, std::mutex> lock(mutex_list_.pose,
+                                                mutex_list_.pointcloud);
```

Comment:
Should I add new topics here every time the topics that I subscribe increase?

Reply:
True! How many topics are you expecting to have?

Reply:
We need two more topics.
1. Information for states of obstacles: in 'ObjectStateArray.msg' type (please refer los_keeper_msgs package)
2. Information for states of targets: in 'ObjectStateArray.msg' type (please refer los_keeper_msgs package)

In addition, we need position, velocity, and acceleration for planning drone trajectory.
Thus, using 'DroneState.msg' (please refer los_keeper_msgs pacakge) instead of 'geometry_msgs::msg::PoseStamped' will be proper for our algorithm.

I think you would have used "geometry_msgs::msg::PoseStamped" when thinking about the drone pose topic from ZED camera. I'm sorry I couldn't tell you earlier.